src/java/org/apache/cassandra/nodes/Nodes.java

```diff
@@ -366,18 +384,16 @@ private void delete(InetAddressAndPort peer, boolean blocking)
             Future<?> f = updateExecutor.submit(wrapPersistenceTask("deleting peer information: " + peer, () -> {
                 nodesPersistence.deletePeer(peer);
                 logger.trace("Deleted peer {}", peer);
-                if (blocking)
-                    nodesPersistence.syncPeers();
             }));
 
             if (blocking)
-                FBUtilities.waitOnFuture(f);
+                FBUtilities.waitOnFuture(f, PERSISTENCE_TIMEOUT);
         }
     }
 
     public class Local
     {
-        private final LoadingMap<InetAddressAndPort, LocalInfo> internalMap = new LoadingMap<>(1);
+        private final LoadingMap<InetAddressAndPort, LocalInfo> internalMap = new LoadingMap<>(1, PERSISTENCE_TIMEOUT.multipliedBy(2));
```

Comment:
PERSISTENCE_TIMEOUT is applied inside the mapping function for the persistence related operations. We don't want the compute function to fail too early - all in all, this is just to prevent undisclosed deadlocks